library/alloc/src/alloc.rs

```diff
@@ -213,25 +205,38 @@ impl Global {
         }
     }
 
+    #[inline]
     #[cfg(not(bootstrap))]
+    fn alloc_untraceable_impl(
+        &self,
+        layout: Layout,
+        _zeroed: bool,
+    ) -> Result<NonNull<[u8]>, AllocError> {
+        match layout.size() {
+            0 => Ok(NonNull::slice_from_raw_parts(layout.dangling(), 0)),
+            // SAFETY: `layout` is non-zero in size,
```

Comment:
I'm not sure what this comment is supposed to be attached to (presumably the second clause?).

Reply:
rustc has a lint which won't allow an unsafe block without a comment explaining why the unsafe block is ok to use.

Reply:
Oh! Would it be clearer to write this as:

```rust
x => y {
   // comment
  unsafe { ... }
}
```

?

Reply:
I agree this is better and I went to change it, but then `x.py fmt` reverted my changes. I think we should leave it because fighting with the formatter is just going to cause pain.

Reply:
If it reverts it, I don't think we have much choice :)